Can not build with clang 11

I got a lot of multiple definition error when I try to build with clang-11

`make CC=clang-11 CXX=clang++-11`

```
commands.o:/home/zyh/redisraft_new/./redismodule.h:568: multiple definition of `RedisModule_ZsetFirstInLexRange'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:568: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:566: multiple definition of `RedisModule_ZsetFirstInScoreRange'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:566: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:562: multiple definition of `RedisModule_ZsetIncrby'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:562: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:569: multiple definition of `RedisModule_ZsetLastInLexRange'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:569: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:567: multiple definition of `RedisModule_ZsetLastInScoreRange'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:567: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:570: multiple definition of `RedisModule_ZsetRangeCurrentElement'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:570: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:573: multiple definition of `RedisModule_ZsetRangeEndReached'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:573: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:571: multiple definition of `RedisModule_ZsetRangeNext'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:571: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:572: multiple definition of `RedisModule_ZsetRangePrev'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:572: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:565: multiple definition of `RedisModule_ZsetRangeStop'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:565: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:564: multiple definition of `RedisModule_ZsetRem'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:564: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:563: multiple definition of `RedisModule_ZsetScore'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:563: first defined here
commands.o:/home/zyh/redisraft_new/./redismodule.h:611: multiple definition of `RedisModule__Assert'
redisraft.o:/home/zyh/redisraft_new/./redismodule.h:611: first defined here
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:66: recipe for target 'redisraft.so' failed
make: *** [redisraft.so] Error 1
```

--- a/redismodule.h
+++ b/redismodule.h
@@ -418,7 +418,7 @@ typedef long long mstime_t;
 #endif
 
 #ifndef REDISMODULE_ATTR_COMMON
-#    if defined(__GNUC__) && !defined(__clang__)
+#    if defined(__GNUC__)
 #        define REDISMODULE_ATTR_COMMON __attribute__((__common__))
 #    else
 #        define REDISMODULE_ATTR_COMMON

--- a/redismodule.h
+++ b/redismodule.h
@@ -418,7 +418,7 @@ typedef long long mstime_t;
 #endif
 
 #ifndef REDISMODULE_ATTR_COMMON
-#    if defined(__GNUC__) && !defined(__cplusplus)
+#    if defined(__GNUC__) && !(defined(__clang) && defined(__cplusplus))
 #        define REDISMODULE_ATTR_COMMON __attribute__((__common__))
 #    else
 #        define REDISMODULE_ATTR_COMMON

--- a/redismodule.h
+++ b/redismodule.h
@@ -418,7 +418,7 @@ typedef long long mstime_t;
 #endif
 
 #ifndef REDISMODULE_ATTR_COMMON
-#    if defined(__GNUC__)
+#    if defined(__GNUC__) && !defined(__cplusplus)
 #        define REDISMODULE_ATTR_COMMON __attribute__((__common__))
 #    else
 #        define REDISMODULE_ATTR_COMMON

--- a/redismodule.h
+++ b/redismodule.h
@@ -418,7 +418,7 @@ typedef long long mstime_t;
 #endif
 
 #ifndef REDISMODULE_ATTR_COMMON
-#    if defined(__GNUC__) && !(defined(__clang) && defined(__cplusplus))
+#    if defined(__GNUC__) && !(defined(__clang__) && defined(__cplusplus))
 #        define REDISMODULE_ATTR_COMMON __attribute__((__common__))
 #    else
 #        define REDISMODULE_ATTR_COMMON

